src/gc-stock.c

```diff
@@ -3511,6 +3547,7 @@ void jl_parallel_gc_threadfun(void *arg)
         if (may_sweep(ptls)) {
             assert(jl_atomic_load_relaxed(&ptls->gc_state) == JL_GC_PARALLEL_COLLECTOR_THREAD);
             gc_sweep_pool_parallel(ptls);
+            sweep_stack_pools();
```

Comment:
Could we introduce another counter in the `ptls` (e.g. `gc_sweep_stack_pools_requested`) to indicate that a sweep of stack pools is pending?

Using `gc_sweeps_requested` for both sweeping of object pools and stack pools makes the code quite hard to reason about, and I'm afraid may not be fully correct as it is.